Makefile

```diff
@@ -40,6 +41,10 @@ doc-json:
 reindent:
 	git ls-files '*.ml*' '**/*.ml*' | xargs ocp-indent --syntax cstruct -i
 
+version:
```

Comment:
Should we let the `build` target depend on this, so that it remains a "do everything" target?

Reply:
I'm not sure about the suggestion, when building from the repository we generally want to use `dune subst` so the version can be grabbed from `git describe`.

I think this target is mainly useful only when building from tarball, when `git describe` is not available.

Reply:
i didn't let the build target depend on version, because you then have to carry around a change in your working directory:

```
[ben@piglet:xen-api {master} (CA-334811)]$ git diff
diff --git a/ocaml/util/xapi_version.ml b/ocaml/util/xapi_version.ml
index 0d903aef8..a67b54072 100644
--- a/ocaml/util/xapi_version.ml
+++ b/ocaml/util/xapi_version.ml
@@ -15,7 +15,7 @@ let (xapi_version_major, xapi_version_minor) =
      - creating an RPM (`make version`)
      - compiling with opam (`dune subst`)
   *)
-  let v = "%%VERSION_NUM%%" in
+  let v = "1.244.0-1-g0303cdaf1" in
   try
     Scanf.sscanf v "%d.%d.%s" (fun maj min _rest -> (maj, min))
   with _ ->
```

Using `dune subst` is actually a lot worse, because it adds a version field to all the opam files...

```
[ben@piglet:xen-api {master} (CA-334811)]$ git status
On branch CA-334811
Your branch is up-to-date with 'origin/CA-334811'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   Makefile
        modified:   dune-project
        modified:   ocaml/util/xapi_version.ml
        modified:   xapi-cli-protocol.opam
        modified:   xapi-client.opam
        modified:   xapi-consts.opam
        modified:   xapi-database.opam
        modified:   xapi-datamodel.opam
        modified:   xapi-types.opam
        modified:   xapi.opam
        modified:   xe.opam

no changes added to commit (use "git add" and/or "git commit -a")
```

Reply:
Yeah, it would be much nicer if the build does not change any tracked files. If it can't be done by environment variables, then perhaps by creating a temporary file that is included (that's how we used to do it).